pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -369,11 +375,11 @@ func (c *Reconciler) resolvePipelineState(
 			}
 			var nfErr *resources.TaskNotFoundError
 			if errors.As(err, &nfErr) {
-				pr.Status.MarkFailed(v1.PipelineRunReasonCouldntGetTask.String(),
+				pr.Status.MarkFailed(!isUserError, v1.PipelineRunReasonCouldntGetTask.String(),
```

Comment:
Thanks Dibyo! There seems to be two questions that needs to be filled in:
1) Shall we use bool or wrapping with a new error?
- Set isUserError as a bool flag, and add it to the pipelineRun.MarkFailed() as a parameter. This will force our contributors to explicitly call out whether this is a user error or not when adding to the reconciler logics.
- Introduce a new error "userError". Wrap the error message with it and at pipelineRun.MarkFailed(), label user using regex to check the error message. However, it is hard to tell if the error introduced is used or not.

2) Do we want to label on TaskRun/PipelineRun reason or message?
After examining the error messages, we have use cases where a TaskRun/PipelineRun reason has many different error messages i.e. PipelineRunReasonFailedValidation. It might be more precise to label error messages.

Reply:
Thanks for the discussion offline.
I think the way forward should be doing both adding a bool flag so that we can leverage judgement from the author and creating a new error type UserError.